addon/utils/decorators/uses-transition.js

```diff
@@ -0,0 +1,14 @@
+import isFastBoot from '../is-fastboot';
+import { assert } from '@ember/debug';
+
+export default function arg(fadeProperty) {
+  assert('You have to provide a fadeProperty for typeClass', typeof fadeProperty === 'string');
+
+  return function () {
+    return {
+      get() {
+        return !isFastBoot(this) && this.args[fadeProperty] !== false;
+      },
+    };
+  };
+}
```

Comment:
I'm not getting how this is related to a transition. It seems to guard against fastboot only. It seems to be a `@unlessFastboot` decorator.

Reply:
It's not really new, we had it already as a CP macro here: https://github.com/kaliber5/ember-bootstrap/blob/master/addon/utils/cp/uses-transition.js

It *does* check for `this.args.fade`, so not really just about FastBoot though!?

Reply:
> It _does_ check for `this.args.fade`, so not really just about FastBoot though!?

Maybe I totally missed something but it doesn't seem to check `this.args.fade` but a dynamic argument passed in. The only relationship to _fade_, which I can see is the variable name `fadeProperty`.

But I agree that we can revisit this later as it was like that before.